crates/core/component/dex/src/component/router/path.rs

```diff
@@ -98,3 +99,30 @@ impl<S: StateRead + 'static> Path<S> {
         }
     }
 }
+
+impl<S: StateRead + 'static> PartialEq for Path<S> {
+    fn eq(&self, other: &Self) -> bool {
+        self.price == other.price && self.nodes == other.nodes
+    }
+}
+
+impl<S: StateRead + 'static> PartialOrd for Path<S> {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl<S: StateRead + 'static> Eq for Path<S> {}
+
+impl<S: StateRead + 'static> Ord for Path<S> {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.price.cmp(&other.price) {
+            Ordering::Equal => match self.nodes.len().cmp(&other.nodes.len()) {
+                Ordering::Greater => Ordering::Greater,
+                Ordering::Equal => self.nodes.cmp(&other.nodes),
+                Ordering::Less => Ordering::Less,
+            },
+            less_or_more => less_or_more,
+        }
+    }
```

Comment:
Could this use `.then_with` to express the sequence of comparisons more succinctly? https://doc.rust-lang.org/stable/std/cmp/enum.Ordering.html#method.then_with